src/Redux/Subscriptions/thunks.js

```diff
@@ -18,98 +18,109 @@ import { handleDisplayToast, setContentToast } from 'Redux/Shared/ResponseToast/
-      dispatch(addSubscriptionsSuccess(data));
-      dispatch(setContentToast({ message: 'Subscription Created', state: 'success' }));
+    } catch (error) {
+      dispatch(getSubscriptionsError(error));
       dispatch(handleDisplayToast(true));
+      dispatch(setContentToast({ message: error.message, state: 'fail' }));
     }
+  };
+};
 
-    if (response.status === 400) {
-      throw new Error('Subscription not created!');
+export const addSubscriptions = (newAddSubscription) => {
+  return async (dispatch) => {
+    dispatch(addSubscriptionsPending());
+    try {
+      const isoDate = newAddSubscription.date
+        ? new Date(newAddSubscription.date).toISOString()
+        : new Date();
+      const body = {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+          token: token
+        },
+        body: JSON.stringify({
+          classes: newAddSubscription.classes,
+          members: newAddSubscription.members,
+          date: isoDate
+        })
+      };
+      const response = await fetch(`${process.env.REACT_APP_API_URL}/api/subscription`, body);
+      const data = await response.json();
+      if (!data.error) {
+        dispatch(addSubscriptionsSuccess(data));
+        dispatch(setContentToast({ message: 'Subscription Created', state: 'success' }));
+        dispatch(handleDisplayToast(true));
+      } else {
+        dispatch(addSubscriptionsError(data.message));
+        dispatch(handleDisplayToast(true));
+        dispatch(setContentToast({ message: data.message, state: 'fail' }));
+      }
+    } catch (error) {
+      dispatch(addSubscriptionsError(error));
+      dispatch(setContentToast({ message: error.message, state: 'fail' }));
+      dispatch(handleDisplayToast(true));
     }
-  } catch (error) {
-    dispatch(addSubscriptionsError(error));
-    dispatch(setContentToast({ message: error.message, state: 'fail' }));
-    dispatch(handleDisplayToast(true));
-  }
+  };
 };
 
-export const editSubscription = async (dispatch, newEditSubscription, id) => {
-  dispatch(editSubscriptionPending());
-  try {
-    const isoDate = newEditSubscription.date
-      ? new Date(newEditSubscription.date).toISOString()
-      : new Date().toISOString;
-    const body = JSON.stringify({
-      classes: newEditSubscription.classes,
-      members: newEditSubscription.members,
-      date: isoDate
-    });
-    const response = await fetch(`${process.env.REACT_APP_API_URL}/api/subscription/${id}`, {
-      method: 'PUT',
-      headers: {
-        'Content-Type': 'application/json',
-        token: token
-      },
-      body: body
-    });
-    const data = await response.json();
-    if (response.status === 200) {
-      dispatch(editSubscriptionSuccess(data));
-      dispatch(setContentToast({ message: 'Subscription edited', state: 'success' }));
+export const editSubscription = (newEditSubscription, id) => {
+  return async (dispatch) => {
+    dispatch(editSubscriptionPending);
```

Comment:
only the parentheses are missing

```suggestion
    dispatch(editSubscriptionPending());
```